pynhm/base/adapter.py

```diff
@@ -68,22 +40,32 @@ def __init__(
         self,
         fname: fileish,
         variable: str,
-        start_time: np.datetime64 = None,
+        control: Control,
     ) -> None:
         super().__init__(variable)
         self.name = "AdapterNetcdf"
+
         self._fname = fname
         self._dataset = NetCdfRead(fname)
         self._datetime = self._dataset.date_times
-        if start_time is None:
-            self._start_time = self._datetime[0]
-        else:
-            self._start_time = start_time
 
-        self._current_value = None
+        if control is None:
+            raise ValueError(f"{self.name} requires a valid Control object.")
+        self.control = control
+        self._start_time = self.control.start_time
+
+        self._nhru = self._dataset.dataset.dimensions["nhm_id"].size
+        self._dtype = self._dataset.dataset.variables[self._variable].dtype
+        self._current_value = np.full(self._nhru, np.nan, self._dtype)
+        # Adopting the next line requires minor changes to most tests
+        # self._current_value = control.get_var_nans(self._variable)
```

Comment:
do the above with a method on control (which requires control to have parameters)
the test changes alluded to involve passing the the params fixture to the control object. but then only passing the control object to the storage units (evenutally).

Reply:
that seemed like a lot of changes if giving params to control was uncertain.